tools/rndf_visualizer/src/MapLanes.cc

```diff
@@ -1151,38 +1170,51 @@ void createEquilateralTriangle(svg::Polygon &triangle, float x, float y, float r
 }
 
 
-void drawWaypoint(svg::Document &doc, float ratio, float min_x, float max_y, float angle, WayPointNode &w1){
+void drawWaypoint(svg::Document &doc, float ratio, float min_x, float max_y, float angle, WayPointNode &w1, bool flat, svg::Fill *defaultColor){
 	static svg::Fill greenFill = svg::Fill(svg::Color::Green);
 	static svg::Fill redFill = svg::Fill(svg::Color::Red);
 	static svg::Fill blueFill = svg::Fill(svg::Color::Blue);
 	static svg::Fill fuchsiaFill = svg::Fill(svg::Color::Fuchsia);
 	static svg::Fill orangeFill = svg::Fill(svg::Color::Orange);
 	static svg::Fill magentaFill = svg::Fill(svg::Color::Magenta);
 	static svg::Fill yellowFill =  svg::Fill(svg::Color::Yellow);
+	static svg::Fill flatFill = svg::Fill(svg::Color::Black);
 
-	if (w1.is_exit) {
-		svg::Polygon arrow = svg::Polygon(redFill);
-		createArrow(arrow, (w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio , 5 * ratio, -1.0*angle);
-		doc.operator << (arrow);		
-	}
-	else if (w1.is_entry) {
-		svg::Polygon arrow = svg::Polygon(blueFill);
-		createArrow(arrow, (w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio , 5 * ratio, -1.0*angle);
-		doc.operator << (arrow);	
-	}
-	else if (w1.is_stop) {
-		doc.operator << (svg::Circle(svg::Point((w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio ), 5 * ratio, orangeFill));
-	}
-	else if (w1.is_perimeter) {
-		doc.operator << (svg::Circle(svg::Point((w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio ), 5 * ratio, yellowFill));
-	}
-	else if (w1.checkpoint_id != 0) {
-		doc.operator << (svg::Circle(svg::Point((w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio ), 5 * ratio, fuchsiaFill));
+	if(flat) {
+		if (w1.is_stop || w1.is_perimeter || w1.checkpoint_id != 0) {
+			doc.operator << (svg::Circle(svg::Point((w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio ), 5 * ratio, *defaultColor));
+		}
+		else {
+			svg::Polygon arrow = svg::Polygon(*defaultColor);
+			createArrow(arrow, (w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio , 5 * ratio, -1.0*angle);
+			doc.operator << (arrow);
+		}
 	}
 	else {
-		svg::Polygon arrow = svg::Polygon(greenFill);
-		createArrow(arrow, (w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio , 5 * ratio, -1.0*angle);
-		doc.operator << (arrow);
+		if (w1.is_exit) {
+			svg::Polygon arrow = svg::Polygon(redFill);
+			createArrow(arrow, (w1.map.x - min_x) * ratio, (max_y - w1.map.y) * ratio , 5 * ratio, -1.0*angle);
+			doc.operator << (arrow);		
+		}
+		else if (w1.is_entry) {
```

Comment:
@basicNew I've seen this. There's a limitation on the SVG library because I don't have setters and getters for Strokes and Fills. This thing headed me to make a method to encapsulate the C&P code. If I make a general "if-else" statement with all the global conditions for circles and/or arrows whose color depends the type of the waypoint I can remove the C&P on figures creation but I will be creating duplicates of the conditions. I think this can remain the same.